Sources/Public/BacktraceBreadcrumbsLogManager.swift

```diff
@@ -0,0 +1,52 @@
+import Foundation
+
+@objc public class BacktraceBreadcrumbsLogManager: NSObject {
+    
+    private var breadcrumbId = Date().millisecondsSince1970
+    
+    private var backtraceQueueFileHelper: BacktraceQueueFileHelper?
+
+    public init(_ breadcrumbLogPath: String, maxQueueFileSizeBytes: Int) {
+        super.init()
+        self.backtraceQueueFileHelper = BacktraceQueueFileHelper(breadcrumbLogPath, maxQueueFileSizeBytes: maxQueueFileSizeBytes)
+    }
+    
+    public func addBreadcrumb(_ message: String,
+                              attributes:[String:Any]? = nil,
+                              type: BacktraceBreadcrumbType,
+                              level: BacktraceBreadcrumbLevel) -> Bool {
+        let time = Date().millisecondsSince1970
+        
+        var info: [String: Any] = ["timestamp": time,
+                    "id": breadcrumbId,
+                    "level": level.info,
+                    "type": type.info,
+                    "message": message]
+        if let attributes = attributes, attributes.keys.count > 0 {
+            var attribInfo: [String: Any] = [String: Any]()
+            for attribute in attributes {
+                attribInfo[attribute.key] = attribute.value
+            }
+            info["attributes"] = attribInfo
+        }
+        
+        if let result = backtraceQueueFileHelper?.addInfo(info), result == true {
+            breadcrumbId += 1
```

Comment:
I would auto increment always, not just when successful

Reply:
See other comment on auto increment - should we auto increment in the beginning instead?